lib/src/widgets/overlay_date_time_picker/overlay.dart

```diff
@@ -0,0 +1,184 @@
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _pageController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            IconButton(
+              onPressed: goToPreviousPage,
+              icon: const Icon(Icons.arrow_circle_left_outlined),
+              color: widget.theme.barTheme.barColor,
+            ),
+            Text(DateFormat.yMMMM().format(
+              widget.controller.browsingDate,
+            )),
+            IconButton(
+              onPressed: goToNextPage,
+              icon: const Icon(Icons.arrow_circle_right_outlined),
+              color: widget.theme.barTheme.barColor,
+            ),
+          ],
+        ),
+        Container(
+          height: 2,
+          decoration: BoxDecoration(
+            color: widget.theme.barTheme.barColor,
+          ),
+        ),
+        Expanded(
+          child: PageView(
+            controller: _pageController,
+            onPageChanged: (value) {
+              if (!usesButtons) movePage(1 - value);
+            },
+            pageSnapping: true,
+            scrollDirection: Axis.horizontal,
+            allowImplicitScrolling: true,
+            children: [
+              DatePicker(
+                controller: widget.controller,
+                onSelectDate: onSelectDate,
+                theme: widget.theme,
+                date: DateTime(
+                  widget.controller.browsingDate.year,
+                  widget.controller.browsingDate.month - 1,
+                  1,
+                ),
+              ),
+              DatePicker(
+                controller: widget.controller,
+                onSelectDate: onSelectDate,
+                theme: widget.theme,
+                date: widget.controller.browsingDate,
+              ),
+              DatePicker(
+                controller: widget.controller,
+                onSelectDate: onSelectDate,
+                theme: widget.theme,
+                date: DateTime(
+                  widget.controller.browsingDate.year,
+                  widget.controller.browsingDate.month + 1,
+                  1,
+                ),
+              ),
+            ],
+          ),
+        )
+      ],
+    );
+  }
+
+  void goToNextPage() {
+    setState(() {
+      usesButtons = true;
+    });
+    _pageController
+        .nextPage(
+      duration: const Duration(milliseconds: 250),
+      curve: Curves.easeInOut,
+    )
+        .then(
+      (value) {
+        nextPage();
+      },
+    );
+  }
+
+  void goToPreviousPage() {
+    setState(() {
+      usesButtons = true;
+    });
+    _pageController
+        .previousPage(
+      duration: const Duration(milliseconds: 250),
+      curve: Curves.easeInOut,
+    )
+        .then(
```

Comment:
Any reason to prefer this over making the entire thing a regular `async` function and awaiting this?